hilti/runtime/include/exception.h

```diff
@@ -24,18 +25,43 @@ class Exception : public std::runtime_error {
     /**
      * @param desc message describing the situation
      */
-    Exception(const std::string& desc) : Exception(Internal(), "Exception", desc) {}
+    Exception(const std::string& desc) : Exception(Internal(), "Exception", desc) {
+#ifndef NDEBUG
+        _backtrace = std::make_unique<Backtrace>();
+#endif
+    }
 
     /**
      * @param desc message describing the situation
      * @param location string indicating the location of the operation that failed
      */
-    Exception(std::string_view desc, std::string_view location) : Exception(Internal(), "Exception", desc, location) {}
+    Exception(std::string_view desc, std::string_view location) : Exception(Internal(), "Exception", desc, location) {
+#ifndef NDEBUG
+        _backtrace = std::make_unique<Backtrace>();
+#endif
+    }
 
     Exception();
-    Exception(const Exception&) = default;
+
+    Exception(const Exception& other)
+        : std::runtime_error(other), _description(other._description), _location(other._location) {
+        if ( other._backtrace )
+            _backtrace = std::make_unique<Backtrace>(*other._backtrace);
+    }
+
     Exception(Exception&&) noexcept = default;
-    Exception& operator=(const Exception&) = default;
+    Exception& operator=(const Exception& other) {
+        // Note: We don't copy the backtrace here, as it's not clear what
+        // that would mean.
```

Comment:
Shouldn't an exception be copyable, including its full state? I cannot see where outside of implementation details we care about the backtrace being unique.